lib/hdprivatekey.js

```diff
@@ -274,7 +296,50 @@ HDPrivateKey.prototype._deriveWithNumber = function(index, hardened, nonComplian
     chainCode: chainCode,
     privateKey: privateKey
   });
+  return derived;
+};
 
+HDPrivateKey.prototype._deriveWithBigNumber = function(index, hardened) {
+  /* jshint maxstatements: 20 */
+  /* jshint maxcomplexity: 10 */
+  if (!HDPrivateKey.isValidPath(index, hardened)) {
+    throw new hdErrors.InvalidPath(index);
+  }
+
+  var indexBuffer = index.toBuffer({size:32}) 
```

Comment:
I would like to point out an issue that we had with the Java version of this function to make sure the JavaScript does not have the same problem.

In Java, we were using `BigInteger` to store the the 256bit value.  If the most significant byte was larger than 0x7F, then this structure stores a leading zero making the buffer 33 bytes.  Then the key derivation is changed.  in Objective C a `uint256` structure is used always ensuring 32 bytes.  With a probability of 50%, payments between iOS and Android generated contact requests would not be received.

I presume the line of code above prevents this from happening along with perhaps the underlying big number structure that is used.